src/main/java/uk/gov/hmcts/ccd/domain/service/caseaccess/CaseAccessOperation.java

```diff
@@ -104,6 +122,15 @@ public void updateUserAccess(CaseDetails caseDetails, CaseUser caseUser) {
             .collect(Collectors.toCollection(ArrayList::new));
     }
 
+    private List<CaseDetails> getCaseDetailsList(List<Long> caseReferences) {
+        return caseReferences.stream()
+            .map(caseReference -> {
+                Optional<CaseDetails> caseDetails = caseDetailsRepository.findByReference(null, caseReference);
```

Comment:
Calling `caseDetailsRepository.findByReference` again and again for every distinct case reference may not be performant enough in for a call sending case role assignments for many distinct cases, but as per the current non-functional requirements, we can still accept this.